### PR TITLE
Add VS unit tests for Navigate to Source

### PR DESCRIPTION
**Pull request by dungpa**

### CODE REVIEW

File: tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+"""
+        let projectFileName = fullPathBasedOnSourceDir "../data/NavigateToSource/NavigateToSource.fsproj"
+        let fileName = fullPathBasedOnSourceDir "../data/NavigateToSource/OctokitTests.fs"
+        let buffer = createMockTextBuffer content fileName
+        helper.SetUpProjectAndCurrentDocument(ExternalProjectProvider(projectFileName), fileName)              
+        let textView = createMockTextView buffer
+        let command = helper.GetCommandFilter(textView)
+        let urlChanged = command.UrlChanged.Value
+        let filter = command :> IOleCommandTarget
+        let prefix = Path.GetFullPath(fileName)
```
Comment (vasily-kirichenko) [inline tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs]: These 9 lines are duplicated in all three tests. Maybe it's a good idea to extract them into a `SetUp` or a plane function?

Comment (vasily-kirichenko) [inline tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs]: `prefix` is unused.

File: tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        
+        testEventTrigger urlChanged "Timed out before being able to go to definition" timeout
+            (fun () -> 
+                textView.Caret.MoveTo(snapshotPoint textView.TextSnapshot 4 11) |> ignore
+                filter.Exec(ref Constants.guidOldStandardCmdSet, Constants.cmdidGoToDefinition, 
+                    0u, IntPtr.Zero, IntPtr.Zero) |> ignore)
+            (fun () -> 
+                command.CurrentUrl
+                |> Option.get
+                |> assertEqual "http://referencesource.microsoft.com/mscorlib/a.html#090eca8621a248ee")
```
Comment (vasily-kirichenko) [inline tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs]:
```fsharp
command.CurrentUrl
|> assertEqual (Some "http://referencesource.microsoft.com/mscorlib/a.html#090eca8621a248ee")
```
?

File: tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        testEventTrigger urlChanged "Timed out before being able to go to definition" timeout
+            (fun () -> 
+                textView.Caret.MoveTo(snapshotPoint textView.TextSnapshot 4 1) |> ignore
+                filter.Exec(ref Constants.guidOldStandardCmdSet, Constants.cmdidGoToDefinition, 
+                    0u, IntPtr.Zero, IntPtr.Zero) |> ignore)
+            (fun () -> 
+                let url = Option.get command.CurrentUrl
+                // We don't assert on hash values since it will be changed on next FAKE release
+                url.Contains("https://raw.github.com/fsharp/FAKE/") |> assertEqual true
+                url.Contains("/src/app/FakeLib/TargetHelper.fs") |> assertEqual true)
```
Comment (vasily-kirichenko) [inline tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs]: `assertTrue`?

File: tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs (hunk 1 above)
Comment (dungpa) [inline tests/FSharpVSPowerTools.Tests/GoToDefinitionFilterCommandTests.fs]: Ok, done.

**Pull request merged**